resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -114,7 +124,7 @@ private[spark] class ExecutorPodsAllocator(
     // both the creation and deletion events. In either case, delete the missing pod
```

Comment:
How often is `onNewSnapshots` called ?
Trying to understand the cost of this method given the changes.

Reply:
this is called by the ExecutorPodsSnapshotsStoreImpl potentially pretty often. Its on start, whenever you set the total executors (which would be dynamic allocation manager updating things), and there is an allocation timer that fires (spark.kubernetes.allocation.batch.delay=1s default) that updates the allocator on any k8s pod changes. We only request small number of pods at a time and iterate, so if you request 20 executors, its definitely going to go through here a few times.

Since it can potentially be called a lot, this is why I made the normal case with no resource profiles a fast path and constructing the resource profile to pod state only when we are using them.  I didn't actually profile that to see how slow it could be but was thinking it still shouldn't be that bad.  The alternative is to store the resource profile to pod state but this required a lot more changes to the snapshot store and lifecycle manager.